### DISCUSSION TITLE
how to extend useClipboard to copy picture(svg or png)?

### DISCUSSION THREAD

**Discussion by ShangChien** [vueuse/vueuse#1606] [Q&A] [answered]
I came across such an [**application**](https://github.com/rdkit/rdkit-structure-renderer) on Github. It allows you to copy images directly from the browser, how is this done? 
![101854](https://user-images.githubusercontent.com/48350636/168715703-70448c17-c6c5-41de-a31d-04275b6dc9fd.png)
I want to implement this with vue, hope to get some advice, thanks！

**Comment by darrylnoakes** [comment, marked as answer]
You can use `navigator.clipboard.write` and specify the MIME type of what you want to copy along with a Blob.

```js
const copy = async (blob) => {
  try {
    await navigator.clipboard.write([
      new ClipboardItem({
        [blob.type]: blob,
      }),
    ]);
  } catch (err) {
    console.error(err.name, err.message);
  }
};
```

This functionality is not implemented in `useClipboard`, but I'm sure someone is willing to. :)

And of course, you can usually view the source code of the website to see how something is done.

See also:
- [MDN docs](https://developer.mozilla.org/en-US/docs/Web/API/Clipboard/write).
- [Web.dev page](https://web.dev/async-clipboard/)
- [Multi-MIME Type Copying with the Async Clipboard API](https://blog.tomayac.com/2020/03/20/multi-mime-type-copying-with-the-async-clipboard-api/)
- [A clipboard magic trick - how to use different MIME types with the Clipboard API](https://www.stefanjudis.com/notes/a-clipboard-magic-trick-how-to-use-different-mime-types-with-the-clipboard/)

**Reply by ShangChien** [reply]
thank you, I will follow your suggestion to try.
